### PR TITLE
MIR-1292 MIR-1293 Updated template datacite2mods.xsl to support ORCID ids given by url

### PR DESCRIPTION
**Pull request by Possommi**
[MIR-1292](https://mycore.atlassian.net/browse/MIR-1292) and [MIR-1293](https://mycore.atlassian.net/browse/MIR-1293).

requires https://github.com/MyCoRe-Org/mycore/pull/2105

[MIR-1292]: https://mycore.atlassian.net/browse/MIR-1292?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MIR-1293]: https://mycore.atlassian.net/browse/MIR-1293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mir-module/src/main/resources/xsl/import/datacite2mods.xsl [hunk 1]
```diff
@@ -39,7 +42,7 @@
             <mods:role>
                 <mods:roleTerm authority="marcrelator" type="code">aut</mods:roleTerm>
             </mods:role>
-            <xsl:apply-templates select="nameIdentifier[@nameIdentifierScheme='ORCID']" />
+            <xsl:apply-templates select="nameIdentifier[string-length(mcrxml:getMatchingString(text(), $orcidRegex)) = (19)]" />
```
Comment (toKrause) [inline mir-module/src/main/resources/xsl/import/datacite2mods.xsl:45]: Not sure if it is wise to ignore the `nameIdentifierScheme`-attribute. Could lead to false positives. 

I'd relax this part to `<xsl:apply-templates select="nameIdentifier" />` (or leave it as is) ...

File: mir-module/src/main/resources/xsl/import/datacite2mods.xsl [hunk 2]
```diff
@@ -53,16 +56,10 @@
         </mods:namePart>
     </xsl:template>
 
-    <xsl:template match="nameIdentifier[@nameIdentifierScheme='ORCID']">
+    <xsl:template match="nameIdentifier">
+        <xsl:variable name="orcid" select="mcrxml:getMatchingString(text(), $orcidRegex)"/>
```
Comment (toKrause) [inline mir-module/src/main/resources/xsl/import/datacite2mods.xsl:60]: ... and just add the new check as an additional precaution here, i.e.

```xml
<xsl:template match="nameIdentifier[@nameIdentifierScheme='ORCID']">
  <xsl:variable name="orcid" select="mcrxml:getMatchingString(text(), $orcidRegex)"/>
  <xsl:if test="string-length($orcid)!=0">
    <mods:nameIdentifier type="orcid">
      <xsl:value-of select="$orcid"/>
    </mods:nameIdentifier>
  </xsl:if>
</xsl:template>
```

Comment (Possommi) [reply mir-module/src/main/resources/xsl/import/datacite2mods.xsl:60]: If I am doing it that way I don't get the ORCIDs imported: 10.5281/zenodo.3987859

Comment (Possommi) [reply mir-module/src/main/resources/xsl/import/datacite2mods.xsl:60]: @toKrause I just checked for the example DOI the @nameIdentifierScheme = 'https://orcid.org' thus the template won't match

**Comment by toKrause** [comment]
The URI used to resolve this DOI is `xslStyle:strip-namespaces,import/datacite2mods:https://data.datacite.org/application/vnd.datacite.datacite+xml/10.5281/zenodo.3987859`.

I think there is a problem in `strip-namespaces.xsl`.

Before that step:

```
    <creator>
      <creatorName>Gerlach, Roman</creatorName>
      <givenName>Roman</givenName>
      <familyName>Gerlach</familyName>
      <nameIdentifier nameIdentifierScheme="ORCID" schemeURI="https://orcid.org">https://orcid.org/0000-0001-5104-4247</nameIdentifier>
      <affiliation>Friedrich-Schiller University of Jena</affiliation>
    </creator>
```

After that step:

```
    <creator>
      <creatorName>Gerlach, Roman</creatorName>
      <givenName>Roman</givenName>
      <familyName>Gerlach</familyName>
      <nameIdentifier>https://orcid.org/0000-0001-5104-4247</nameIdentifier>
      <affiliation>Friedrich-Schiller University of Jena</affiliation>
    </creator>
```

**Comment by Possommi** [comment]
> The URI used to resolve this DOI is `xslStyle:strip-namespaces,import/datacite2mods:https://data.datacite.org/application/vnd.datacite.datacite+xml/10.5281/zenodo.3987859`.
> 
> I think there is a problem in `strip-namespaces.xsl`.
> 
> Before that step:
> 
> ```
>     <creator>
>       <creatorName>Gerlach, Roman</creatorName>
>       <givenName>Roman</givenName>
>       <familyName>Gerlach</familyName>
>       <nameIdentifier nameIdentifierScheme="ORCID" schemeURI="https://orcid.org">https://orcid.org/0000-0001-5104-4247</nameIdentifier>
>       <affiliation>Friedrich-Schiller University of Jena</affiliation>
>     </creator>
> ```
> 
> After that step:
> 
> ```
>     <creator>
>       <creatorName>Gerlach, Roman</creatorName>
>       <givenName>Roman</givenName>
>       <familyName>Gerlach</familyName>
>       <nameIdentifier>https://orcid.org/0000-0001-5104-4247</nameIdentifier>
>       <affiliation>Friedrich-Schiller University of Jena</affiliation>
>     </creator>
> ```

Updated strip-namespaces.xsl and this fixes the issue with both ORCID and abstracts

**Comment by toKrause** [comment]
> Updated strip-namespaces.xsl and this fixes the issue with both ORCID and abstracts

As per the documentation, `node()` should already select all nodes (["`node()` matches any node. "](https://www.w3.org/TR/xpath-31/#id-matching-item)). 

Not sure why this is necessary (`<xsl:apply-templates select="@* | node()"/>` ), but I came to the same conclusion. Without the `@*`, it just doesn't work. Confusing, ...

**Pull request merged**